flexbe_onboard/src/flexbe_onboard/flexbe_onboard.py

```diff
@@ -336,8 +342,31 @@ def _execute_heartbeat(self):
     def _heartbeat_worker(self):
```

Comment:
`_heartbeat_worker` is called from a separate thread. You should protect the members that are accessed from `produce_diagnostics`

Reply:
not sure how you would do it, but using the available `self._switch_lock` or `self._run_lock` are never released from during SM execution, thus the `heartbeat` and `diagnostics` are never published....because they never get the locks...

Reply:
Even with read-only access, a variable must be protected if it is written by another thread. Imagine that the memory where the variable is located is accessed by the writing and reading thread at the same time => Bang :-)

As we do only have one reading and one writing thread you could simply protect the members that are accessed with a `threading.Lock`. Before reading it acquire the lock and when reading/coping the value is done release the lock. Before changing the value acquire the lock and when finished release the lock.